crates/sui/src/client_ptb/ast.rs

```diff
@@ -0,0 +1,353 @@
+                if let Some(v) = o {
+                    let v = v.as_ref().to_pure_move_value(*loc).map_err(|e| e.with_help(
+                            format!(
+                                "Was unable to parse '{self}' as a pure PTB value. This is most likely because \
+                                the option contains a non-primitive (e.g., object or array) \
+                                value which isn't permitted inside an option"
+                                )
+                            ))?;
+                    MoveValue::Vector(vec![v])
+                } else {
+                    MoveValue::Vector(vec![])
+                }
+            }
+            Argument::Identifier(_)
+            | Argument::VariableAccess(_, _)
+            | Argument::Gas => error!(loc, "Was unable to convert '{self}' to primitive value (i.e., non-object value)"),
+        })
+    }
+}
+
+impl fmt::Display for Argument {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Argument::Bool(b) => write!(f, "{b}"),
+            Argument::U8(u) => write!(f, "{u}u8"),
+            Argument::U16(u) => write!(f, "{u}u16"),
+            Argument::U32(u) => write!(f, "{u}u32"),
+            Argument::U64(u) => write!(f, "{u}u64"),
+            Argument::U128(u) => write!(f, "{u}u128"),
+            Argument::U256(u) => write!(f, "{u}u256"),
+            Argument::Gas => write!(f, "gas"),
+            Argument::Identifier(i) => write!(f, "{i}"),
+            Argument::VariableAccess(sp!(_, head), accesses) => {
+                write!(f, "{}", head)?;
+                for sp!(_, access) in accesses {
+                    write!(f, ".{}", access)?;
+                }
+                Ok(())
+            }
+            Argument::Address(a) => write!(f, "@{a}"),
+            Argument::String(s) => write!(f, "{s:?}"),
+            Argument::Vector(v) => {
+                write!(f, "vector[")?;
+                let mut prefix = "";
+                for sp!(_, arg) in v.iter() {
+                    write!(f, "{prefix}")?;
+                    write!(f, "{arg}")?;
+                    prefix = ", ";
+                }
+                write!(f, "]")
+            }
+            Argument::Option(sp!(_, o)) => match o {
+                Some(v) => write!(f, "some({v})"),
+                None => write!(f, "none"),
+            },
+        }
+    }
+}
+
+fn delimited_list<T: fmt::Display>(
+    f: &mut std::fmt::Formatter<'_>,
+    sep: &str,
+    items: impl IntoIterator<Item = T>,
+) -> std::fmt::Result {
+    let mut prefix = "";
+    for item in items {
+        write!(f, "{}{}", prefix, item)?;
+        prefix = sep;
+    }
+    Ok(())
+}
+
+impl fmt::Display for ParsedPTBCommand {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ParsedPTBCommand::TransferObjects(arg, args) => {
+                write!(f, "{TRANSFER_OBJECTS} {} [", arg.value)?;
```

Comment:
It translates to `transfer-objects {} [ args_values ]` (this is referring to the const further above). I'm not sure if we want to add the `--` in front though, since it might make the output a bit noisy?

Reply:
Oh, I misread, sorry. All good!